Launch a login shell by default on Linux

gentooboontoo opened the pull request:
Alacritty did not launch a login shell by default on Linux contrary to
what is claimed by the comments in the configuration file. As a result,
`bash` does not read `/etc/profile` file leading to an inconsistent
environment compared to those with other terminals.

This commit simply extends what has been done by PR #1856 referenced by
issue #1071.

NickeZ commented:
This does not seem to be the defualt behavior on `gnome-terminal`. The following should print 0 if it is a login shell.

```
$ shopt -q login_shell; echo $?
1
```

gentooboontoo commented:
You are right, it is different between my system and other pristine systems (I wrongly assumed it was the same).

I was misled because I always use `tmux` inside my terminals and `tmux` launch shells as login shells for new windows. Without `tmux`, my other installed terminals (`gnome-terminal`, `terminator`) also launch login shells but I should have configured them long time ago and I forgot that (so it does not match anymore their default configuration).

My questions now are:

* Should MacOS and linux `alacritty` terminals' behavior be consistent?
* If not, why is it different on MacOS? (out of curiosity, I have no Mac)
* If yes, I could reword the commit description removing the part about other terminals' behavior.
* If this PR is not the right thing to do, I could close it and make another one fixing the comment about default login shells in [`alacritty.conf` (line 366)](https://github.com/jwilm/alacritty/blob/v0.3.3-rc1/alacritty.yml#L366).

chrisduerr commented:
> Should MacOS and linux alacritty terminals' behavior be consistent?

No, the reason why `--login` is explicitly required on macOS is because otherwise the environment will be messed up and the locale is not set correctly. Linux does not have this issue.

> If this PR is not the right thing to do, I could close it and make another one fixing the comment about default login shells in alacritty.conf

I don't think it's necessary to open a new PR, but that seems reasonable. The current docs are probably incorrect because there has been some refactorings in the past, Alacritty will not use `bash` by default on Linux at all for example.

gentooboontoo commented:
> Alacritty will not use bash by default on Linux at all for example

I missed that point.

I have reverted my change and fixed the docs about default shell. I am unsure about how format the description of the linux default shell since it is not a "static" default value.

When and if OK, I will squash the commits and rename the title of the PR.

chrisduerr commented on alacritty.yml:
```diff
@@ -363,7 +363,8 @@ live_config_reload: true
 #
 # Default:
-#   - (Linux/macOS) /bin/bash --login
+#   - (macOS) /bin/bash
```
The --login argument was correct for macOS, it should be kept.

chrisduerr commented on alacritty.yml:
```diff
@@ -363,7 +363,8 @@ live_config_reload: true
 # Default:
-#   - (Linux/macOS) /bin/bash --login
+#   - (macOS) /bin/bash
+#   - (Linux) shell program defined in password file for current user
```
I think this can be reduced to just `user login shell`, that should make things clear since it uses common terminology.

chrisduerr commented:
> When and if OK, I will squash the commits and rename the title of the PR.

No need to, I squash and fix the commit message of every PR when merging anyways.

gentooboontoo commented:
Requested changes applied.
